osbs/api.py

```diff
@@ -351,25 +352,30 @@ def create_prod_build(self, git_uri, git_ref,
         :param git_ref: str, reference to commit
         :param git_branch: str, branch name (may be None)
         :param user: str, user name
-        :param component: str, component name
-        :param target: str, koji target (may be None)
+        :param component: str, not used anymore
+        :param target: str, koji target
         :param architecture: str, build architecture
         :param yum_repourls: list, URLs for yum repos
         :param koji_task_id: int, koji task ID requesting build
         :param scratch: bool, this is a scratch build
         :return: BuildResponse instance
         """
+
+        def get_required_label(labels, names):
+            for label_name in names:
+                if label_name in labels:
+                    return labels[label_name]
+            
+            raise OsbsValidationException("required label '{name}' missing "
+                                          "from Dockerfile"
+                                          .format(name=label_name))
```

Comment:
I'm always wary about using a variable that is defined in a for-loop statement.
If names is empty, the for loop will not set `label_name` which would cause this statement to raise something like: `NameError: name 'label_name' is not defined`

This is a nested method so it's unlikely to hit this scenario, but if this is a concern, checks name `names` is truthy and raise a `ValueError` with a meaningful message.